### PR TITLE
GDB-7681: Initial implementation of component

### PR DESCRIPTION
**Pull request by boyan-tonchev**
What?

 Added a configurator for base yasgui configuration.
Why?

 As clients of component we need an opportunity to configure endpoints and headers of yasgui component.
How?

 Implements base building flow of a yasgui instance.
 Implements builder of yasgui configuration.
 Implements a configuration which will configure base configuration of yasgu.

### CODE REVIEW

File: cypress/e2e/home/home-page.spec.cy.ts [hunk 1]
```diff
@@ -2,8 +2,15 @@ import PageSteps from '../../steps/page-steps';
...
 describe('Describes "ontotext-yasgui-web-component" component', () => {
 
+   beforeEach(() => {
+      cy.intercept('/repositories/test-repo', {fixture: '/queries/default-query-response.json'}).as('getGuides');
+   });
+
    it('Should load "ontotext-yasgui-web-component" component', () => {
       PageSteps.visitHomePage();
       cy.getByDataSelector('ontotext-yasgui-tag');
+      cy.get('.yasqe_queryButton').click();
```
Comment (svilenvelikov) [inline cypress/e2e/home/home-page.spec.cy.ts:12]: We should move these in the steps to abstract the view access

File: ontotext-yasgui-web-component/src/services/yasgui/configuration/configurator.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ */
+export interface Configurator {
+
+  // @ts-ignore
+  /**
+   * Apply configuration to <code>yasguiConfig</code> from <code>clientYasguiConfig</code>
+   * @param yasguiConfig - configuration which is passing to yasgui constructor.
+   * @param customYasguiConfig - configuration passed by client.
+   */
+  config(yasguiConfig, customYasguiConfig: CustomYasguiConfiguration);
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/services/yasgui/configuration/configurator.ts:15]: Where is the type of the `yasguiConfig` parameter?

File: ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configurator.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        'Accept': 'application/sparql-results+json',
+        'X-GraphDB-Local-Consistency': 'updating'
+      };
+    }
+  }
+
+  config(yasguiConfig, customYasguiConfig: CustomYasguiConfiguration) {
+    const result = {... yasguiConfig, ... this.defaultYasguiConfig};
+    result.requestConfig = {... this.defaultRequestConfig}
+    result.requestConfig.endpoint = customYasguiConfig.endpoint;
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configurator.ts:29]: Why don't just extend the request config with the provided one. SHould we add here each and every possible parameter one by one?

File: cypress/e2e/home/home-page.spec.cy.ts (hunk 1 above)
Comment (boyan-tonchev) [reply cypress/e2e/home/home-page.spec.cy.ts]: done

File: ontotext-yasgui-web-component/src/components/ontotext-yasgui/ontotext-yasgui.tsx [hunk 4]
```diff
@@ -40,7 +47,7 @@ export class OntotextYasgui {
   }
 
   @Watch('config')
-  configurationChanged(newConfig: YasguiConfiguration) {
+  configurationChanged(newConfig: CustomYasguiConfiguration) {
```
[the remark this answers is not in the document]
Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/components/ontotext-yasgui/ontotext-yasgui.tsx]: done

File: ontotext-yasgui-web-component/src/services/yasgui/configuration/configurator.ts (hunk 2 above)
Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/services/yasgui/configuration/configurator.ts]: done

File: ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configurator.ts (hunk 3 above)
Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configurator.ts]: done

**Comment by boyan-tonchev** [comment]
Jenkins retry build

File: ontotext-yasgui-web-component/src/services/yasgui/yasgui-builder.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    const yasguiConfig = this.yasguiConfigurationBuilder.build(config);
+    // @ts-ignore
+    const yasgui = new Yasgui(el, yasguiConfig);
+
+    // monkey patches have to be applied before return yasgui.
+    return yasgui;
+  }
+}
+
+export const YaguiBuilder = new YasguiBuilderDefinition();
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/services/yasgui/yasgui-builder.ts:33]: typo in YaguiBuilder

**Review by svilenvelikov — APPROVED** [review]

File: ontotext-yasgui-web-component/src/services/yasgui/yasgui-builder.ts (hunk 5 above)
Comment (boyan-tonchev) [reply ontotext-yasgui-web-component/src/services/yasgui/yasgui-builder.ts]: fixed

**Pull request merged**